snuba/datasets/transactions_processor.py

```diff
@@ -297,15 +297,37 @@ def _process_spans(
             ):
                 return
 
+            num_processed = 0
             processed_spans = []
 
             processed_root_span = self._process_span(trace_context)
             if processed_root_span is not None:
+                num_processed += 1
                 processed_spans.append(processed_root_span)
 
             for span in data.get("spans", []):
+                # The number of spans should not exceed 1000 as enforced by SDKs.
+                # As a safety precaution, enforce a hard limit on the number of
+                # spans we actually store .
+                if num_processed >= settings.MAX_SPANS_PER_TRANSACTION:
```

Comment:
Let's make this a runtime flag so we can change it quickly if/when needed. With a default of 2000. We will keep the default most of the time.